syntax = "proto3";

import "google/protobuf/empty.proto";
import "vertex.proto";

package generated;

option optimize_for = LITE_RUNTIME;
option cc_enable_arenas = true;

message pcl_data {
	repeated vertex_3d vertices = 1;
	int64 timestamp = 2;
}

message draco_data {
	bytes data = 1;
	int64 timestamp = 2;
}

message maybe_matrix {
	optional matrix data = 1;
}

service pcl_com {

  //sends point clouds and receives possible correspondence
  rpc transmit_pcl_data (stream pcl_data) returns (maybe_matrix) {}
  rpc transmit_draco_data (stream draco_data) returns (maybe_matrix) {}
  rpc transmit_obb (obb) returns (google.protobuf.Empty) {}
}